### DISCUSSION TITLE
19 Ideas including suckless to terminal stuffs

### DISCUSSION THREAD

**Discussion by kneardhead** [xournalpp/xournalpp#3259] [Ideas]
I was looking for some good alternative to microsoft one note, something opensourced like Blender or Krita. 

Xournal++ was as close it could get, it's amazing, but it wasn't near one note. 

Now there's hardly any such projects. I thought because we are talking bout Opensource, let's try to implement the ideas in xournal++. 

1 >\_ A very handy screen snip support. 
2 >\_ Because I have a graphics tab pen on the right hand, making the left hand side of keyboard do the most of stuffs.
3 >\_ There's a bug that new pages are inserted between the pages, not at the last end. It might even be something I am doing wrong. I have to recheck, it might not be a bug. But what about a shortcut way to insert pages in the middle?
4 >\_ A floating tool, triggered when a button on the tab pen is pressed. It's already in experimental phase, but If it's like Krita and round, it would be cooler
5 >\_ The gui needs to be worked out a little, there's some unnecessary stuffs that can be manipulated if the key binding is made. A more minimal (kinda) (?) 
6 >\_ Some hand writing math to latex math tool 
7 >\_ Vim interface 
8 >\_ More colorscheme themes ( :slight_smile: www.reddit.com/r/unixporn/ )
9 >\_ Little more "suckless" (suckless.org) philosophy made, so lighter and terminal like interface (like unixporn) 
10>\_ Transpiracy effect? (another pdf can be kept open behind so transparecy effect will be epic. 
11 >\_ Pywal integration will be nice
12 >\_ Packages, like atom does or like python modules 
13 >\_ The PDF output is kinda trashy, it needs a fix
14 >\_ Infinite writing interface mode (it currently works on only paper size) 
15 >\_ Some touch of Minimalism is needed. 
16 >\_ (How about Krita integration :3 ) 
17 >\_ Jupyter addon (can be good) 
18 >\_ Cloud or at least github repository/ google drive integration addon 
19 >\_ Not to update these epic changes for the windows port for free XD (just kidding) 

I would like to fork it and try to implement it myself to suit the needs. I am a random physics student and but I can do tinker with the forked code. If you are aware, **please please please** share some resource so that I can study it and try to edit the source code to work out these idea. Yes, I do know Cpp, but I don't have any idea how to use it here.

I am newbie and I highly appreciate your suggestions.

**Comment by rolandlo** [comment]
Thanks for shearing your ideas. What OS are you using?

About 1> What would you expect of a screen snip tool? There is the `QuickScreenshot` plugin for some minimal functionality.

About 2> You are free to define your keybindings that way by using a plugin. See also [this vi-style plugin](https://github.com/raw-bacon/vi-xournalpp)

About 4> This could probably be implemented as a Lua plugin. 

About 3> When you add a page via the toolbar, it will be added after the current page. There is also a "New Page at End" in menu `Journal`. Not sure why you see a bug here.

About 5> and 15> You can create a minimal custom toolbar and enter fullscreen mode (or simply hide the menubar and sidebar) to get some minimalism.

About 13> I don't know why you consider PDF export to be trashy. It may depend on how recent your libraries (in particular Cairo) are.

Things like 14> and 18> are in planning stage, but are quite complex.

I think you should focus on one thing and then we could help you get started with it (if it is realistic enough).

**Reply by kneardhead** [reply]
I am in Ubuntu, groovy gorilla. 

Reply to 1> Umm, something like, using the right hand we press, say ``T`` and the cursor becomes a screen snip tool, then navigating to the place, we confirm the region of screensnip by drawing that selection zone and confirming, with, say ``T`` again (just something I have in mind) 

2 > Thanks, didn't know about it, I really appreciate. 

4> Hmm, okay I kinda get it. 

3> Yeah, I guess I doing something wrong, nvm.

5> 15> yeah I get it. But I was thinking something more basic like, say, umm, a Terminal (?) ? 

13> Yeah, the thickness of line becomes the same. But it's not a big issue tho, I export the png and it works really good. 

yeah I understand the thing. Thanks a whole lot for replying I really appreciate! 



Well, I'd like to know how the app really works to be honest, I love working with it, but it would awesome to look under the hood. 

I'd appreciate if you _link me to something that I can study and have an idea how the software is working/or how it's made/ basically what is happening under the hood._ 

And anyway, thanks for your cordial reply and plug-in suggestions, it would really help.

**Reply by rolandlo** [reply]
> Reply to 1> Umm, something like, using the right hand we press, say `T` and the cursor becomes a screen snip tool, then navigating to the place, we confirm the region of screensnip by drawing that selection zone and confirming, with, say `T` again (just something I have in mind)

Have you tried the QuickScreenshot Plugin? You must activate it in the plugin manager (and restart Xournal++). You press the key-combination (which you can modify in the plugin) or use menu `Plugin`. The cursor becomes a screen snip tool at once and you can select the selection zone by clicking and releasing the mouse. When finished, you are prompted to save the file on your system. If you'd prefer to put it into the clipboard (and maybe paste it back into your Xournal++ document), then that would be even quite a bit simpler.

**Reply by kneardhead** [reply]
I am yet to try when I get a bit more time (I have a 4 hour exam today xD), 
after it I promise to try it out and the check the code thoroughly.

**Reply by kenballus** [reply]
I would say having a screenshot utility built into Xournal++ is pretty much the opposite of suckless, if that's what you're into. There are plenty of great screenshot utilities out there already, no need to reinvent the wheel in my opinion. Try flameshot.

**Reply by kneardhead** [reply]
yes you're right. Thanks for the suggestion! I am trying it.

**Comment by rolandlo** [comment]
If you want to look under the hood and study some code which is easy to modify, then I would advise you to look at the `CoordinateSystemHandler`, see [code](https://github.com/xournalpp/xournalpp/blob/master/src/control/tools/CoordinateSystemHandler.cpp)

If you haven't built Xournal++ from source yet, see [wiki](https://github.com/xournalpp/xournalpp/blob/master/readme/LinuxBuild.md)

**Reply by kneardhead** [reply]
That's great, thank you!

**Comment by personalizedrefrigerator** [comment]
Suggestion 10 is implemented by #3269!

**Reply by kneardhead** [reply]
That's so cool, I am going to try it asap.

**Comment by j-lakeman** [comment]
> 8 >_ More colorscheme themes ( :slight_smile: [www.reddit.com/r/unixporn/](http://www.reddit.com/r/unixporn/) )

https://github.com/xournalpp/xournalpp/discussions/6537
